functions.php

```diff
@@ -0,0 +1,41 @@
+<?php
+    /**
+    * Creates a connection to the collectorapp database
+    */
+    function createDBConnection() :PDO{
+        $db = new PDO('mysql:host=db; dbname=collectorapp', 'root', 'password');
+        $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+        return $db;
+    }
+
+    /**
+    * A query to the database that fetches the data in all the fields and creates a recipes assoc array
+    * @return array
+    */
+    function fetchAllRecipes($db): array {
+        $query = $db->prepare('SELECT `recipe`, `cuisine`, `time`, `link` FROM `recipes`;');
+        $query->execute();
+        $recipes = $query->fetchAll();
+        return $recipes;
+    }
+
+    /**
+    * Creates recipe cards which have a name as well as stats on cuisine, time and a link to the recipe
+    */
+    function createRecipeCards(array $recipes) {
+        if (count($recipes)>0) {
+            $recipeCards = '';
+            foreach ($recipes as $recipe) {
+                $recipeCards .= '<section class= "recipe_card">';
+                $recipeCards .= "<h2>{$recipe['recipe']}</h2>";
+                $recipeCards .= "<h3>Cuisine: {$recipe['cuisine']}</h3>";
+                $recipeCards .= "<h3>Time (mins): {$recipe['time']}</h3>";
+                $recipeCards .= "<a href= {$recipe['link']}>Link to recipe</a>";
+                $recipeCards .= "</section>";
+            }
+            return $recipeCards;
+        } else {
+            $noRecipesError = 'No available recipes';
+            return $noRecipesError;
```

Comment:
In this case I would advise just returning the string. 

You only really need to save something in a variable if you're going to do something else with it, in this case just writing `return 'No available recipes';` will do the trick.

It's incredibly minor, there would be a tiny tiny tiny performance improvement, but we're talking reallllllllllly tiny like 0.00000000001 seconds